src/Data/LLVM/BitCode/GetBits.hs

```diff
@@ -26,91 +30,79 @@ import qualified Control.Monad.Fail
 
 -- Bit-level Parsing -----------------------------------------------------------
 
-newtype GetBits a = GetBits { unGetBits :: Either String (BitPosition -> BS.ByteString -> (BitsGetter a, BitPosition)) } -- Left is outer fail
+newtype GetBits a =
+  GetBits { unGetBits :: BitPosition -> BS.ByteString
+                      -> (# BitsGetter a, BitPosition #)
+          }
```

Comment:
This removes the outer fail by getting rid of the `Either`. Does this mean that the `Either` in `BitsGetter` handles both forms of failure now?